Mathlib/RingTheory/LocalProperties/Flat.lean

```diff
@@ -0,0 +1,214 @@
+/-
+Copyright (c) 2024 Sihan Su. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: SiHan Su
+-/
+import Mathlib.RingTheory.Flat.Stability
+import Mathlib.RingTheory.LocalProperties.LinearMap
+/-!
+# Flat is a local property
+
+In this file, we prove that `Module.Flat` is a local property.
+
+## Main Results
+
+* `Flat_of_localization_maximal` : Let `M` be a module over `CommRing R`. If the localization
+  of `M` at each maximal ideal `P` is flat over `R_p`, then `M` is flat over `R`.
+* `Flat_of_localization_finitespan` : Let `M` be a module over `CommRing R` and `S` be a set that
+  spans `R`. If the localization of `M` at `s : S` is flat over `Localization.Away s`,
+  then `M` is flat over `R`.
+* `flat_iff_localization` : Let `R_p` a localization of `CommRing R` and `M` be a module over `R_p`.
+  Then `M` is flat over `R` if and only if `M` is flat over `R_p`.
```

Comment:
Since you're not using LaTeX, it's better to change R_p to Rₚ.
```suggestion
* `Flat_of_localization_maximal` : Let `M` be a module over `CommRing R`. If the localization
  of `M` at each maximal ideal `P` is flat over `R_p`, then `M` is flat over `R`.
* `Flat_of_localization_finitespan` : Let `M` be a module over `CommRing R` and `S` be a set that
  spans `R`. If the localization of `M` at each `s : S` is flat over `Localization.Away s`,
  then `M` is flat over `R`.
* `flat_iff_localization` : Let `R_p` a localization of `CommRing R` and `M` be a module over `R_p`.
  Then `M` is flat over `R` if and only if `M` is flat over `R_p`.
```